scraper/spiders/citycouncil.py

```diff
@@ -86,87 +113,76 @@ def get_status(status):
         status = strip_accents(status.strip())
         return status.lower().replace(" ", "_")
 
-    def parse(self, response):
-        boxes = response.css("div.row div div")
-        current_page = response.meta.get("current_page", 1)
-        last_page = response.css("ul.pagination li:last-child ::text").get()
-        found = False
-
-        for box in boxes:
-            list_date = box.css("ul li ::text").get()
-            if list_date:
-                date_obj = parse(list_date, dayfirst=True)
-                if date_obj.date() >= self.start_date:
-                    found = True
-                    list_url = box.css("div a::attr(href)").get()
-                    yield scrapy.Request(
-                        url=response.urljoin(list_url),
-                        callback=self.parse_list_page,
-                        meta={"date": list_date},
-                    )
-
-        if found and last_page:  # deve continuar checando até não encontrar mais
-            last_page = int(last_page)
-            next_page = current_page + 1
-            if next_page <= last_page:
-                yield scrapy.Request(
-                    url=response.urljoin(f"listadepresenca.asp?p={next_page}"),
-                    callback=self.parse,
-                    meta={"current_page": next_page},
-                )
-
     def parse_list_page(self, response):
-        council_members = response.css("div.row div div ul li a::text").extract()
-        status = response.css("div.row div div div a::text").extract()
-
-        for council_member, status in zip(council_members, status):
-            yield CityCouncilAttendanceListItem(
-                crawled_at=datetime_utcnow_aware(),
-                crawled_from=response.url,
-                date=from_str_to_date(response.meta["date"]),
-                council_member=council_member.strip(),
-                status=self.get_status(status),
-            )
+        dates = response.css("div#myTabContent2 h4::text").extract()
+        tables = response.css("table.table")
+
+        for a_date in dates:
+            table = tables.pop()
+            info = table.css("div.TITULO::text").extract()
+            council_members = []
+            status = []
+            index = 0
+            while index < len(info):
+                if index % 2 == 0:  # even
+                    council_members.append(info[index])
+                else:
+                    status.append(info[index])
+                index += 1
+
+            for council_member, status in zip(council_members, status):
+                yield CityCouncilAttendanceListItem(
+                    crawled_at=datetime_utcnow_aware(),
+                    crawled_from=response.url,
+                    date=from_str_to_date(a_date),
```

Comment:
Algumas datas estão retornando nulo e falhando na hora de salvar no banco de 🎲 

```
{'council_member': 'Zé Curuca',
 'crawled_at': datetime.datetime(2023, 7, 3, 20, 57, 59, 461601, tzinfo=datetime.timezone.utc),
 'crawled_from': 'https://www.feiradesantana.ba.leg.br/lista-presenca?mes=05&ano=2023&Acessar=OK',
 'date': None,
 'description': 'Sessão Ordinária',
 'git_commit': '',
 'status': 'presente'}
```

As vezes isso acontece porque uma página em específico muda 🆘 talvez resultado de uma atualização manual. Eu testei se o problema era com a ambiguidade entre dia e mês, já que 2 e 5 podem ser dias ou meses, mas não era esse o problema.

Pra só testar a partir desse mês, pode usar o comando:
```
SPIDER=citycouncil_attendancelist START_DATE=2023-05-01 make runspider
```

Colocando um print no spider, vi que a data vem vazia e aí quando passamos pra o método que faz a conversão ele retorna `None`. Aí fui tentando ver qual a data começava a falhar e era no mês de maio 🤔 printando somente as datas deu pra ver que o mês de maio tá meio doidão no html:

```
['04/04/2023',
 '05/04/2023',
 '11/04/2023',
 '12/04/2023',
 '13/04/2023',
 '18/04/2023',
 '19/04/2023',
 '25/04/2023',
 '26/04/2023',
 '27/04/2023']
>>>>>>>>>>>>>>>>>>>>
['02/05/2023',
 '03/05/2023',
 '04/05/2023',
 '09/05/2023',
 '10/05/2023',
 '11/05/2023',
 '16/05/2023',
 '        ',
 '17/05/2023',
 '        ',
 '18/05/2023',
 '        \r\n        ',
 '23/05/2023',
 '        ']
```

Aí temos que ver se podemos simplesmente pular as datas que retornam vazio com um `continue` sem perder nenhum dado (acho que o erro, na verdade, acontece pq tem um `<br>` largado nessas últimas datas aff). Confirmando que conseguimos raspar tudo dá pra usar um `a_date.strip()` e não executar o bloco de código se a data estiver vazia.

Me avisa se deu pra entender! Fui investigando e anotando. 😅

Reply:
Ah, e se for isso mesmo, temos que verificar nos outros códigos que usam essa mesma estrutura (se usarem).

Reply:
Boa! Fez sentido, vou verificar!

Reply:
Os que vem vazio precisam ser retirados pois estavam ficando no indice da data correta. Fiz a funcao que corrige.

Reply:
Tambem percorri as datas dos outros spiders e nenhuma outra tem espaco em branco.